Sync_app/moysklad/moysklad_class_lib.py

```diff
@@ -370,12 +372,31 @@ class MoySklad:
     # https://dev.moysklad.ru/doc/api/remap/1.2/#mojsklad-json-api-obschie-swedeniq-autentifikaciq
     _token: str = ""
 
+    @staticmethod
+    def _exclude_returned_goods(
+            sold_goods: list[RetailDemandPosition], returned_goods: list[RetailReturnedPosition],
+    ) -> list[RetailDemandPosition]:
+        """Метод удаляет количество товара из списка returned_goods, в списке sold_goods."""
+        sold_goods_dict = {good.good_id: good for good in sold_goods}
+        for ret in returned_goods:
+            if sold_goods_dict.get(ret.good_id):
+                if sold_goods_dict[ret.good_id].quantity - ret.quantity < 1:
+                    del sold_goods_dict[ret.good_id]
+                else:
+                    sold_goods_dict[ret.good_id] = sold_goods_dict[ret.good_id]._replace(
+                        quantity=(sold_goods_dict[ret.good_id].quantity - ret.quantity),
+                    )
```

Comment:
О, тут крч можно заюзать новомодную штуку walrus operator
```suggestion
            if sold_good := sold_goods_dict.get(ret.good_id):
                if sold_good.quantity - ret.quantity < 1:
                    del sold_goods_dict[ret.good_id]
                else:
                    sold_goods_dict[ret.good_id] = sold_good._replace(
                        quantity=(sold_good.quantity - ret.quantity),
                    )
```